apps/smp-server/Main.hs

```diff
@@ -1,18 +1,71 @@
+{-# LANGUAGE OverloadedStrings #-}
+
 module Main where
 
+import Control.Monad (when)
+import Data.Attoparsec.ByteString.Char8 (Parser)
+import qualified Data.ByteString.Char8 as B
+import Data.Char (toLower)
+import qualified Simplex.Messaging.Crypto as C
+import Simplex.Messaging.Parsers (parseAll)
 import Simplex.Messaging.Server (runSMPServer)
 import Simplex.Messaging.Server.Env.STM
+import System.Directory (doesFileExist, getAppUserDataDirectory)
+import System.Exit (exitFailure)
+import System.FilePath (combine)
+import System.IO (hFlush, stdout)
 
 cfg :: ServerConfig
 cfg =
   ServerConfig
     { tcpPort = "5223",
       tbqSize = 16,
       queueIdBytes = 12,
-      msgIdBytes = 6
+      msgIdBytes = 6,
+      -- keys are loaded from files server_key.pub and server_key in ~/.simplex directory
+      serverKeyPair = undefined
     }
 
+newKeySize :: Int
+newKeySize = 2048 `div` 8
+
 main :: IO ()
 main = do
-  putStrLn $ "Listening on port " ++ tcpPort cfg
-  runSMPServer cfg
+  (k, pk) <- readCreateKeys
+  B.putStrLn $ "SMP transport key hash: " <> publicKeyHash k
+  putStrLn $ "Listening on port " <> tcpPort cfg
+  runSMPServer cfg {serverKeyPair = (k, pk)}
+
+readCreateKeys :: IO C.KeyPair
+readCreateKeys = do
+  dir <- getAppUserDataDirectory "simplex"
```

Comment:
put these into /etc/simplex similar to ssh?

https://www.ssh.com/ssh/host-key

Reply:
or better /etc/opt/simplex according to this
https://www.pathname.com/fhs/pub/fhs-2.3.html#OPTADDONAPPLICATIONSOFTWAREPACKAGES